datascience/tables.py

```diff
@@ -1238,12 +1238,31 @@ def join(self, column_label, other, other_label=None):
         3    | 2    | 4
         3    | 2    | 5
         1    | 10   | 6
+        >>> table.join(['a', 'b'], table2, ['a', 'd']) # joining on multiple columns
+        a    | b    | c    | e
+        1    | 10   | 6    | 6
+        9    | 1    | 3    | 3
         """
         if self.num_rows == 0 or other.num_rows == 0:
             return None
         if not other_label:
             other_label = column_label
 
+        # checking to see if joining on multiple columns
+        if _is_non_string_iterable(column_label):
+            # then we are going to be joining multiple labels
+            return self._multiple_join(column_label, other, other_label)
+
+        # original single column join
+        return self._join(column_label, other, other_label)
+
+    def _join(self, column_label, other, other_label=None):
+        """joins when COLUMN_LABEL is a string"""
+        if self.num_rows == 0 or other.num_rows == 0:
+            return None
```

Comment:
What's the purpose of the check in lines 1261-1262?  Do we need this?  I would have expected that if you try to join an empty table to some other table, it would be more intuitive to return an empty table rather than None.  Is that right?  I'm not sure we should be changing the behavior of .join() on an empty table -- that seems like a change that could potentially break prior code.